### ISSUE TITLE
inconsistent behavior when using -C -G and -W

### ISSUE THREAD

**Issue by aieri** [the-tcpdump-group/tcpdump#695]
### Example1
**Command:** `tcpdump -i eth0 -w test.pcap -G10 -C100 -W2`
**Expected result:** test.pcap0 and test.pcap1, either smaller than 100MB, rotation every 10 seconds
**Actual result:** test.pcap0 is overwritten every 10 seconds

### Example2
**Command:** `tcpdump -i eth0 -w test-%s.pcap -G10 -C100 -W2`
**Expected result:** 2x test-\<epoch>.pcap0, rotation every 10 seconds (and .pcap1 if more than 100MB is captured within the 10 seconds)
**Actual result:** an unlimited number of test-\<epoch>.pcap0

Basically, the -W flag is ignored in both cases. Not using it results in the same behavior, minus the "0" suffix.

Can be observed in the current master branch as well.

**Comment by agnosticdev** [comment]
I am not sure the `-W` flag is being ignored, but possibly the `-C100` could be causing an issue.

I am able to reproduce the result in example 1 and see the `test.pcap` file being overwritten instead of iterated upon.  Removing the `-C100` argument from this example does eliminate this issue.  That being said, I am unable to produce a 100 mb file in 10 seconds to accurately test this case.

I am able to reproduce the case in example 2 as well.  Removing the `-C100` argument and changing command to the following below might get you close to where you want to be in the meantime:
```tcpdump -i en0 -G 10 -w trace-%S.pcap -W2```

**Comment by aieri** [comment]
Indeed if you remove the -C flag, things work as expected:
`tcpdump -i en0 -G 10 -w trace-%S.pcap -W2` produces two files, each 10 seconds long, with tcpdump exiting after 20 seconds.

However, I wanted to have an infinitely rotating buffer that is time based, and not size based (removing the `-G` option achieves that).

The documentation states for flag -W:

>       -W     Used in conjunction with the -C option, this will limit the number  of  files
>              created  to the specified number, and begin overwriting files from the begin‐
>              ning, thus creating a 'rotating' buffer.  In addition, it will name the files
>              with  enough leading 0s to support the maximum number of files, allowing them
>              to sort correctly.
>
>              Used in conjunction with the -G option, this will limit the number of rotated
>              dump  files  that get created, exiting with status 0 when reaching the limit.
>              If used with -C as well, the behavior  will  result  in  cyclical  files  per
>              timeslice.

The first two combinations (`-W` + `-C` and `-W` + `-G`) work properly, but the last one doesn't for me (`-W` + `-G` + `-C`).

**Comment by agnosticdev** [comment]
@aieri it does look like things are behaving as expected based upon the documentation.  My interpretation of the documentation, and please anyone correct me if I am wrong, is that:

> Used in conjunction with the -G option, this will limit the number of rotated
         dump  files  that get created, exiting with status 0 when reaching the limit.
         If used with -C as well, the behavior  will  result  in  cyclical  files  per
         timeslice.

is that this cycles the same file buffer being written based upon the `-G` time slice.

**Comment by aieri** [comment]
@agnosticdev: better, but if we are amending the documentation, I would propose something more explicit. Please see the proposed PR.
I still think it would be nice to have a way to set up a cyclical time-based capture, but that can perhaps be a separate feature request.

**Comment by agnosticdev** [comment]
Thank you @aieri!  The PR looks good.  Just one minor suggestion.

**Comment by aieri** [comment]
Oh boy, I gave this another look and my assessment was incorrect (this is really pretty confusing...). The presence of %s in the filename specification does matter, but it already does so when only using **-C** and **-G**, and exactly in the same way: since filenames are evaluated at every timeslice, capture truncation occurs whenever there's a name clash.
The presence of the **-W** flag when both **-C** and **-G** are present is really just cosmetic: filenames get an appended number of zeros depending on the number of files indicated with **-W**, even though that limit will never be reached.

I have updated the PR accordingly.

**Comment by agnosticdev** [comment]
Wanted to check back on this issue to see if there was anything that needs a second look?
There is a PR open for this issue: https://github.com/the-tcpdump-group/tcpdump/pull/722
Thank you.

**Comment by aieri** [comment]
I am of course satisfied with the rewording of the documentation. Not sure if someone from the core team wants to chip in though.
